### ISSUE TITLE
 Cannot read properties of undefined (reading 'getTime')

### ISSUE THREAD

**Issue by alsherif-khalaf** [RSC-Labs/medusa-store-analytics#6]
Just installed it as docs , when click on All time, it give this error :

![image](https://github.com/RSC-Labs/medusa-store-analytics/assets/73169053/7285a463-d124-4d43-b9d4-3a901d6887e9)
![image](https://github.com/RSC-Labs/medusa-store-analytics/assets/73169053/d056954c-3e79-49b4-adff-043a50066a34)

```bash

::ffff:127.0.0.1 - - [24/Mar/2024:19:10:17 +0000] "GET /admin/products-analytics/returned-by-count? HTTP/1.1" 304 - "http://localhost:9000/app/a/analytics" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0"
::ffff:127.0.0.1 - - [24/Mar/2024:19:10:17 +0000] "GET /admin/products-analytics/sold-count?orderStatuses%5B0%5D=completed&orderStatuses%5B1%5D=pending HTTP/1.1" 304 - "http://localhost:9000/app/a/analytics" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0"
::ffff:127.0.0.1 - - [24/Mar/2024:19:10:17 +0000] "GET /admin/marketing-analytics/discounts-by-count?orderStatuses%5B0%5D=completed&orderStatuses%5B1%5D=pending HTTP/1.1" 304 - "http://localhost:9000/app/a/analytics" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0"
error:   Cannot read properties of undefined (reading 'getTime')
TypeError: Cannot read properties of undefined (reading 'getTime')
    at CustomersAnalyticsService.getNewCount (/home/alsherif/New_Dev/medusa-backend/node_modules/@rsc-labs/medusa-store-analytics/dist/services/customersAnalytics.js:153:43)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async GET (/home/alsherif/New_Dev/medusa-backend/node_modules/@rsc-labs/medusa-store-analytics/dist/api/admin/customers-analytics/[kind]/route.js:35:22)
::ffff:127.0.0.1 - - [24/Mar/2024:19:10:18 +0000] "GET /admin/customers-analytics/count? HTTP/1.1" 500 86 "http://localhost:9000/app/a/analytics" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0"
::ffff:127.0.0.1 - - [24/Mar/2024:19:10:19 +0000] "GET /admin/batch-jobs?limit=100 HTTP/1.1" 304 - "http://localhost:9000/app/a/analytics" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0"
::ffff:127.0.0.1 - - [24/Mar/2024:19:10:29 +0000] "GET /admin/batch-jobs?limit=100 HTTP/1.1" 304 - "http://localhost:9000/app/a/analytics" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0"

```

**Comment by radoslaw-sz** [comment]
hi @alsherif-khalaf , thank you for raising this issue and provided logs. Indeed, it is a bug. I will correct it, I think tomorrow or day after tomorrow.

**Comment by radoslaw-sz** [comment]
hi @alsherif-khalaf  - fix landed in version v0.10.2. Please kindly check if it works :)

**Comment by alsherif-khalaf** [comment]
Works like a charm  
![image](https://github.com/RSC-Labs/medusa-store-analytics/assets/73169053/c3b435ff-a2c3-450f-9a77-86f713267c1d)
